syslog/syslog.c

```diff
@@ -0,0 +1,375 @@
+/*
+ * syslog.c
+ *
+ *
+ * Copyright 2015 Susi's Strolch
+ *
+ * For license information see projects "License.txt"
+ *
+ */
+
+#include <esp8266.h>
+#include "config.h"
+#include <syslog.h>
+#include <time.h>
+#include "task.h"
+
+#define SYSLOG_DBG
+#ifdef SYSLOG_DBG
+#define DBG_SYSLOG(format, ...) os_printf(format, ## __VA_ARGS__)
+#else
+#define DBG_SYSLOG(format, ...) do { } while(0)
+#endif
+
+LOCAL os_timer_t check_udp_timer;
+LOCAL struct espconn syslog_espconn;
+LOCAL int syslog_msgid = 1;
+LOCAL uint8_t syslog_task = 0;
+
+LOCAL syslog_host_t	syslogHost;
+LOCAL syslog_entry_t *syslogQueue = NULL;
+
+static enum syslog_state syslogState = SYSLOG_NONE;
+
+LOCAL void ICACHE_FLASH_ATTR syslog_add_entry(char *msg);
+LOCAL void ICACHE_FLASH_ATTR syslog_check_udp(void);
+LOCAL void ICACHE_FLASH_ATTR syslog_udp_sent_cb(void *arg);
+#ifdef SYSLOG_UDP_RECV
+LOCAL void ICACHE_FLASH_ATTR syslog_udp_recv_cb(void *arg, char *pusrdata, unsigned short length);
+#endif
+
+#define syslog_send_udp() post_usr_task(syslog_task,0)
+
+/******************************************************************************
+ * FunctionName : syslog_check_udp
+ * Description  : check whether get ip addr or not
+ * Parameters   : none
+ * Returns      : none
+*******************************************************************************/
+LOCAL void ICACHE_FLASH_ATTR
+syslog_check_udp(void)
+{
+    struct ip_info ipconfig;
+    DBG_SYSLOG("syslog_check_udp: state: %d ", syslogState);
+
+   //disarm timer first
+    os_timer_disarm(&check_udp_timer);
+
+   //get ip info of ESP8266 station
+    wifi_get_ip_info(STATION_IF, &ipconfig);
+    if (wifi_station_get_connect_status() == STATION_GOT_IP && ipconfig.ip.addr != 0)
+    {
+    if (syslogState == SYSLOG_WAIT) {			// waiting for initialization
+    DBG_SYSLOG("connected, initializing UDP socket\n");
+    syslog_init(flashConfig.syslog_host);
+    }
+    } else {
+    DBG_SYSLOG("waiting 100ms\n");
+    if ((wifi_station_get_connect_status() == STATION_WRONG_PASSWORD ||
+    wifi_station_get_connect_status() == STATION_NO_AP_FOUND ||
+    wifi_station_get_connect_status() == STATION_CONNECT_FAIL)) {
+    os_printf("*** connect failure!!! \r\n");
+    } else {
+    //re-arm timer to check ip
+    os_timer_setfn(&check_udp_timer, (os_timer_func_t *)syslog_check_udp, NULL);
+    os_timer_arm(&check_udp_timer, 100, 0);
```

Comment:
Usually (at least with my router) Wifi is up after approx. 2.5s. So, a 
10s increment would be a bit rough.
The 1s sounds like a compromize.

On 11/24/2015 07:01 AM, Thorsten von Eicken wrote:
>
> In syslog/syslog.c 
> <https://github.com/jeelabs/esp-link/pull/73#discussion_r45699483>:
>
> > +    if (wifi_station_get_connect_status() == STATION_GOT_IP && ipconfig.ip.addr != 0)
> > +    {
> > +    if (syslogState == SYSLOG_WAIT) {			// waiting for initialization
> > +    DBG_SYSLOG("connected, initializing UDP socket\n");
> > +    syslog_init(flashConfig.syslog_host);
> > +    }
> > +    } else {
> > +    DBG_SYSLOG("waiting 100ms\n");
> > +    if ((wifi_station_get_connect_status() == STATION_WRONG_PASSWORD ||
> > +    wifi_station_get_connect_status() == STATION_NO_AP_FOUND ||
> > +    wifi_station_get_connect_status() == STATION_CONNECT_FAIL)) {
> > +    os_printf("*** connect failure!!! \r\n");
> > +    } else {
> > +    //re-arm timer to check ip
> > +    os_timer_setfn(&check_udp_timer, (os_timer_func_t *)syslog_check_udp, NULL);
> > +    os_timer_arm(&check_udp_timer, 100, 0);
>
> Does it make sense to check every 100ms? Seems like checking every 1 
> or even 10 seconds would be fine? Maybe it doesn't matter...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jeelabs/esp-link/pull/73/files#r45699483>.
>